packages/plugin-legacy/index.js

```diff
@@ -248,7 +248,7 @@ function viteLegacyPlugin(options = {}) {
               targets,
               modules: false,
               bugfixes: true,
-              loose: true,
+              loose: false,
```

Comment:
**suggestion (idea):** Don't know if it's useful in anycase or why we need to set loose mode to `false` here, but on some places we already often have the possibility to let users decide to override such options :thinking: 
So we may do something like `loose: options.babelTransformOptions?.loose ?? false` here, could we?
_Maybe there is also a better name than_ `babelTransformOptions`

Reply:
Yeah, but perhaps it's better to wait for someone to explicitly request it? Feels like you should use loose mode only in specific situations (i.e. your own code, where you know that you don't need the stricter mode).

But since you don't know what syntax every dependency uses, it's unlikely to be a good idea to transpile your entire bundle through loose mode.

Reply:
`loose` mode equals `false` seems to generate much more output code and if you don't use this `[...new Set()]` in any way, you are not affected by it but now forced by Vite to live with this more generated output for other stuff :thinking:  
Also you can workaround it by not using `[...new Set()]` but `[].concat(Array.from(new Set()))` + using a polyfill for `Array.from`

Reply:
I know that it generates a bigger bundle, but this error comes from Vue itself. So that's a pretty common case for Vite users, and not something you can work around like in your own code.

Reply:
With `loose: options.babelTransformOptions?.loose ?? false` `false` would be the new default, nothing against this
But we let users that are not in this specific context decide and prevent regression in any case for them if they want to
It's downwards compatible / non-breaking with an option
And later we may also use it for e.g. let users override any options with using `{ /*...*/, ...(options.babelTransformOptions ?? {}) }`

So all in all, it's just the possibility to not break anything if there are any breaking changes we currently oversee

Reply:
Maybe it's `babelTransformEnvOptions`?